coderd/database/queries/oauth2.sql

```diff
@@ -60,3 +63,77 @@ WHERE id = $1 RETURNING *;
 
 -- name: DeleteOAuth2ProviderAppSecretByID :exec
 DELETE FROM oauth2_provider_app_secrets WHERE id = $1;
+
+-- name: GetOAuth2ProviderAppCodeByID :one
+SELECT * FROM oauth2_provider_app_codes WHERE id = $1;
+
+-- name: GetOAuth2ProviderAppCodeByAppIDAndSecret :one
+SELECT * FROM oauth2_provider_app_codes WHERE app_id = $1 AND hashed_secret = $2;
+
+-- name: InsertOAuth2ProviderAppCode :one
+INSERT INTO oauth2_provider_app_codes (
+    id,
+    created_at,
+    expires_at,
+    hashed_secret,
+    app_id,
+    user_id
+) VALUES(
+    $1,
+    $2,
+    $3,
+    $4,
+    $5,
+    $6
+) RETURNING *;
+
+-- name: DeleteOAuth2ProviderAppCodeByID :exec
+DELETE FROM oauth2_provider_app_codes WHERE id = $1;
+
+-- name: DeleteOAuth2ProviderAppCodesByAppAndUserID :exec
+DELETE FROM oauth2_provider_app_codes WHERE app_id = $1 AND user_id = $2;
+
+-- name: GetOAuth2ProviderAppTokenByAppSecretIDAndSecret :one
+SELECT * FROM oauth2_provider_app_tokens WHERE app_secret_id = $1 AND hashed_secret = $2;
+
+-- name: InsertOAuth2ProviderAppToken :one
+INSERT INTO oauth2_provider_app_tokens (
+    id,
+    created_at,
+    expires_at,
+    hashed_secret,
+    app_secret_id,
+    api_key_id
+) VALUES(
+    $1,
+    $2,
+    $3,
+    $4,
+    $5,
+    $6
+) RETURNING *;
+
+-- name: GetOAuth2ProviderAppsByUserID :many
+SELECT COUNT(DISTINCT oauth2_provider_app_tokens.id) as token_count,
+  oauth2_provider_apps.id,
+  oauth2_provider_apps.name,
+  oauth2_provider_apps.icon,
+  oauth2_provider_apps.callback_url
```

Comment:
Use a struct embed here.
```suggestion
SELECT COUNT(DISTINCT oauth2_provider_app_tokens.id) as token_count,
  sqlc.embed(oauth2_provider_apps)
```

Reply:
Ooo this is nice.